test/jdk/java/util/prefs/AddNodeChangeListener.java

```diff
@@ -49,28 +50,57 @@ public static void main(String[] args)
         //Should initate a child removed event
         removeNode();
 
-        if (failed)
+        if (failed) {
             throw new RuntimeException("Failed");
+        }
     }
 
     private static void addNode()
-        throws BackingStoreException, InterruptedException
-    {
+            throws BackingStoreException, InterruptedException {
         N2 = userRoot.node("N2");
         userRoot.flush();
-        Thread.sleep(3000);
-        if (ncla.getAddNumber() != 1)
-            failed = true;
+        int passItr = -1;
+
+        for (int i = 0; i < SLEEP_ITRS; i++) {
+            Thread.sleep(3000);
```

Comment:
Would there be any value in printing the iteration number here and in the analogous location in removeNode() in case the test times out?

Reply:
That's a pretty good idea.  In theory, there should be a maximum of 90s total of sleeping, and enough time for the test to complete within the default timeout of 2m.  But AFAICT in the failure case, the test takes longer to run, so maybe it could timeout in that case.  If this failure eventually shows up again, it would be best to have at least *some* new output to show for it. :)